### DISCUSSION TITLE
How do log requests in React Router 7?

### DISCUSSION THREAD

**Discussion by MoSattler** [remix-run/react-router#12666] [Q&A]
Whats the best way to log each request in react-router 7? Without having to reveal certain files or eject in any way?

**Comment by gustavopch** [comment]
I'd say you're pretty much expected to reveal the entry files and have your own server in any repo that's not a super simple demo. It's nothing like ejecting from CRA. Don't be afraid.

**Reply by MoSattler** [reply]
Is there any documentation on how to go about this?

**Reply by gustavopch** [reply]
Use Express and log requests as you normally would: https://github.com/remix-run/react-router/tree/main/playground/framework-express
